internal/commands/push/command.go

```diff
@@ -80,26 +80,63 @@ func (cmd *Command) Handler(profile *cli.Profile, ui terminal.UI, clients cli.Cl
 	}
 
 	ui.Print(terminal.NewTextLog("Determining changes"))
-	diffs, err := clients.Realm.Diff(to.GroupID, to.AppID, app.AppData)
+	appDiffs, err := clients.Realm.Diff(to.GroupID, to.AppID, app.AppData)
 	if err != nil {
 		return err
 	}
 
-	if cmd.inputs.IncludeHosting {
-		// TODO(REALMC-7177): diff hosting changes
-		diffs = append(diffs, "Import hosting")
+	hosting, err := local.FindAppHosting(app.RootDir)
+	if err != nil {
+		return err
 	}
 
-	if cmd.inputs.IncludeDependencies {
-		diffs = append(diffs, "+ New function dependencies")
+	var hostingDiffs local.HostingDiffs
+	if cmd.inputs.IncludeHosting {
+		appAssets, err := clients.Realm.HostingAssets(to.GroupID, to.AppID)
+		if err != nil {
+			return err
+		}
+
+		hostingDiffs, err = hosting.Diffs(profile.HostingAssetCachePath(), to.AppID, appAssets)
+		if err != nil {
+			return err
+		}
 	}
 
-	if len(diffs) == 0 {
+	if len(appDiffs) == 0 && !cmd.inputs.IncludeDependencies && hostingDiffs.Size() == 0 {
 		ui.Print(terminal.NewTextLog("Deployed app is identical to proposed version, nothing to do"))
 		return nil
 	}
 
 	if !ui.AutoConfirm() && !isNewApp {
+		diffs := make([]string, 0, len(appDiffs)+1+hostingDiffs.Size()+3)
+
+		diffs = append(diffs, appDiffs...)
```

Comment:
do we need to create a new slice here or can we just use appDiffs?

Reply:
i didn't want to mutate `appDiffs` because we will check it further downstream when deciding if we need to do the `create draft` => `import` => `deploy app` flow (independent of the dependencies or hosting flows)

Reply:
i didn't want to mutate `appDiffs` because we will check it further downstream when deciding if we need to do the `create draft` => `import` => `deploy app` flow (independent of the dependencies or hosting flows)